module/x/gravity/keeper/batch_test.go

```diff
@@ -428,4 +431,48 @@ func TestGetUnconfirmedBatchTxs(t *testing.T) {
 
 	require.Empty(t, gk.GetUnsignedBatchTxs(ctx, val1))
 	require.Equal(t, 1, len(gk.GetUnsignedBatchTxs(ctx, val2)))
+
+	// Confirm ordering
+	val3, err := sdk.ValAddressFromBech32(vals[2].OperatorAddress)
+	require.NoError(t, err)
+
+	addressA := "0xAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
+	addressB := "0xBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB"
+	gk.SetCompletedOutgoingTx(ctx, &types.BatchTx{
+		TokenContract: addressB,
+		BatchNonce:    3,
+		Height:        blockheight,
+	})
+	gk.SetCompletedOutgoingTx(ctx, &types.BatchTx{
+		TokenContract: addressA,
+		BatchNonce:    4,
+		Height:        blockheight,
+	})
+	gk.SetOutgoingTx(ctx, &types.BatchTx{
+		TokenContract: addressB,
+		BatchNonce:    5,
+		Height:        blockheight,
+	})
+	gk.SetOutgoingTx(ctx, &types.BatchTx{
+		TokenContract: addressA,
+		BatchNonce:    6,
+		Height:        blockheight,
+	})
+	gk.SetOutgoingTx(ctx, &types.BatchTx{
+		TokenContract: addressB,
+		BatchNonce:    7,
+		Height:        blockheight,
+	})
+
+	unconfirmed := gk.GetUnsignedBatchTxs(ctx, val3)
+	fmt.Printf("%v", unconfirmed)
+	require.Equal(t, 7, len(unconfirmed))
+	require.True(t,
```

Comment:
Can't you just check for specific values rather than this cascading series of comparisons?